File: client/src/app/core/services/credit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {environment} from "../../../environments/environment";
import {Credit} from "../models/credit.model";

@Injectable({
  providedIn: 'root',
})
export class CreditService {
  private apiUrl = `${environment.apiUrl}/credits`;

  constructor(private http: HttpClient) {}

  getAllCredits(): Observable<Credit[]> {
    return this.http.get<Credit[]>(this.apiUrl);
  }

  getCreditById(id: number): Observable<Credit> {
    return this.http.get<Credit>(`${this.apiUrl}/${id}`);
  }

  getCreditsByClientId(clientId: number): Observable<Credit[]> {
    return this.http.get<Credit[]>(`${this.apiUrl}/client/${clientId}`);
  }

  createCredit(credit: Credit): Observable<Credit> {
    return this.http.post<Credit>(this.apiUrl, credit);
  }

  updateCredit(id: number, credit: Credit): Observable<Credit> {
    return this.http.put<Credit>(`${this.apiUrl}/${id}`, credit);
  }

  deleteCredit(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: server/src/main/java/com/jeraidi_yassir/server/services/CreditService.java
package com.jeraidi_yassir.server.services;

import com.jeraidi_yassir.server.dtos.ClientDTO;
import com.jeraidi_yassir.server.dtos.CreditDTO;

import java.util.List;

public interface CreditService {
    List<CreditDTO> getAllCredits();

    CreditDTO getCreditById(Long creditId);

    List<CreditDTO> getCreditsByClientId(Long clientId);

    ClientDTO createClient(ClientDTO clientDTO);

    CreditDTO createCredit(CreditDTO creditDTO);

    CreditDTO updateCredit(Long creditId, CreditDTO creditDTO);

    void deleteCredit(Long creditId);
}
